### ISSUE TITLE
TextArea crash in 1.9.9-SNAPSHOT with FreeType

### ISSUE THREAD

**Issue by EtK2000** [libgdx/libgdx#5204]
Please ensure you have given all the following requested information in your report.

#### Issue details
Since updating from 1.9.8 to 1.9.9-SNAPSHOT I've been having a crash caused by a TextArea.
TextFields work properly for me.
This only happens when using a font created using FreeType.
The "times" font I'm using is the Windows builtin Times New Roman font (C:/Windows/Fonts/times.ttf).

#### Reproduction steps/code
```java
public class ErrTest extends ApplicationAdapter {
	private FreeTypeFontGenerator g32;
	private BitmapFont u32;
	private Skin skin;
	private Stage stage;

	@Override
	public void create() {
		g32 = new FreeTypeFontGenerator(Gdx.files.internal("times.ttf"));
		FreeTypeFontParameter param = new FreeTypeFontParameter();
		param.gamma = 1;
		param.incremental = true;
		param.magFilter = TextureFilter.Linear;
		param.minFilter = TextureFilter.Linear;
		param.renderCount = 1;
		param.size = 32;
		u32 = g32.generateFont(param);
		u32.getData().markupEnabled = true;

		skin = new Skin(Gdx.files.internal("uiskin.json"));
		skin.get(TextFieldStyle.class).font = u32;
		stage = new Stage();

		TextArea ta = new TextArea("some text\ncontaining\nmultiple lines", skin);
		ta.setSize(stage.getWidth(), stage.getHeight());
		stage.addActor(ta);
		
		Gdx.input.setInputProcessor(stage);
	}

	@Override
	public void render() {
		stage.act(Gdx.graphics.getDeltaTime());
		stage.draw();
	}

	@Override
	public void dispose() {
		u32.dispose();
		((FreeTypeBitmapFontData) u32.getData()).dispose();
		g32.dispose();
		stage.dispose();
		skin.dispose();
	}
}
```
click the textarea to set it as the stage's focused actor to throw the exception.
I threw this together quickly so it's possible the dispose() might cause an issue or two.

#### Version of LibGDX and/or relevant dependencies
GDX and FreeType: 1.9.9-SNAPSHOT

#### Stacktrace
```java
Exception in thread "LWJGL Application" java.lang.ArrayIndexOutOfBoundsException: 21
	at com.badlogic.gdx.scenes.scene2d.ui.TextArea.letterUnderCursor(TextArea.java:86)
	at com.badlogic.gdx.scenes.scene2d.ui.TextField$TextFieldClickListener.setCursorPosition(TextField.java:842)
	at com.badlogic.gdx.scenes.scene2d.ui.TextArea$TextAreaListener.setCursorPosition(TextArea.java:391)
	at com.badlogic.gdx.scenes.scene2d.ui.TextField$TextFieldClickListener.touchDown(TextField.java:822)
	at com.badlogic.gdx.scenes.scene2d.InputListener.handle(InputListener.java:57)
	at com.badlogic.gdx.scenes.scene2d.Actor.notify(Actor.java:182)
	at com.badlogic.gdx.scenes.scene2d.Actor.fire(Actor.java:147)
	at com.badlogic.gdx.scenes.scene2d.Stage.touchDown(Stage.java:280)
	at com.badlogic.gdx.backends.lwjgl.LwjglInput.processEvents(LwjglInput.java:329)
	at com.badlogic.gdx.backends.lwjgl.LwjglApplication.mainLoop(LwjglApplication.java:217)
	at com.badlogic.gdx.backends.lwjgl.LwjglApplication$1.run(LwjglApplication.java:126)

```

#### Please select the affected platforms
- [x] Android
- [ ] iOS (robovm)
- [ ] iOS (MOE)
- [ ] HTML/GWT
- [x] Windows
- [x] Linux
- [ ] MacOS
(only tested these, probably all)

**Comment by EtK2000** [comment]
I've updated the code to follow that standard.
I guess this needs to be reopened now.

**Comment by NathanSweet** [comment]
Thanks.

When GlyphLayout was added, the old BitmapFont methods for laying out glyphs were replaced. At that time TextField was modified:
3ba1176e112278b7e20f13f2d0178576a5f061b6

The code is was replaced with only looks at the first run. If there are newlines, there are multiple runs but `glyphPositions` only contains entries for the first run and you get an exception, as you found.

I will commit the easiest fix as a stopgap. TextArea is one of the few widgets I did not contribute. I have never needed it in my apps and so I hope others can maintain it. It is maybe a bit odd in that TextField uses GlyphLayout for layout and drawing, while TextArea does its own drawing, line by line. TextArea extends TextField and customizes a few things. Maybe that code reuse is nice, but it probably also makes it a bit fragile.
